src/Type/Php/GetoptFunctionDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\TypeUtils;
+
+class GetoptFunctionDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'getopt';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		return TypeUtils::toBenevolentUnion(TypeCombinator::union(
+			ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType(),
+			new ConstantBooleanType(false)
```

Comment:
You don't have to `union` with `false` here as it's already there in the functionMap.

Reply:
I was under the impression that the return type extension overrode the function map type entirely.